test/unit/test_inference.py

```diff
@@ -476,6 +476,42 @@ def test_topk_func(batch_size, beam_size, target_vocab_size):
     assert all(mx_values == np_values)
 
 
+@pytest.mark.parametrize("batch_size, beam_size, target_vocab_size, top_n",
+                        [(1, 5, 200, 0),
+                         (5, 5, 200, 0),
+                         (1, 100, 200, 5),
+                         (5, 100, 200, 5)])
+def test_samplek_func(batch_size, beam_size, target_vocab_size, top_n):
+    # arrange scores increasing values from left to right, so the best item is always index 0, next-best 1, and so on
+    scores = mx.nd.array([list(range(1, target_vocab_size + 1)) for x in range(batch_size * beam_size)])
+    # normalize
+    target_dists = mx.nd.broadcast_div(scores, mx.nd.sum(scores, axis=1, keepdims=True))
+
+    samplek = sockeye.inference.SampleK(k=beam_size, n=top_n, batch_size=batch_size, context=mx.context.cpu())
+    samplek.initialize()
+
+    # 0..(batch_size * beam_size)-1
+    expected_hyps = mx.nd.array(range(batch_size * beam_size), dtype='int32')
+    finished = mx.nd.array(mx.nd.random.uniform(0, 1, (batch_size * beam_size)) > 0.5, dtype='int32')
+
+    for i in [1, 2]:
```

Comment:
These are independent iterations right? Note that when you hybridize a block i staya hybridized, so if you want to test hybridized block and non-hybridized then I would only hybridize the second iteration.

Reply:
Argh—I had `range(2)` and then made the loop values explicit without updating the conditional. Thanks for the catch!